### DISCUSSION TITLE
VSC plugin for hxc files?

### DISCUSSION THREAD

**Discussion by macohi** [FunkinCrew/Funkin#3860] [Q&A]
Does anyone have a VSC plugin I can use to make hscript scripting easier or nah?

**Comment by moxie-coder** [comment]
Just use the haxe plugin, Hscript is just haxe code

**Reply by macohi** [reply]
Yeah but with hxc files it doesnt give auto-completion, doesn't tell me errors, it treats hxc files as it's own thing even if I set the language setting for the hxc files to haxe

**Reply by charlesisfeline** [reply]
> Yeah but with hxc files it doesnt give auto-completion, doesn't tell me errors, it treats hxc files as it's own thing even if I set the language setting for the hxc files to haxe

dont forget to mention the lack of formatting

**Reply by NotHyper-474** [reply]
> it doesnt give auto-completion, doesn't tell me errors

Thing is, the Haxe/vshaxe will refuse to provide any of that for files that don't have .hx or .hxml as their extension:
`[Warn  - 21:55:50] Exception while processing notification textDocument/didOpen: file:///test.hxc has unsupported file type (must be .hx or .hxml)`

And even if you temporarily rename the file to .hx it might even give some auto-complete and formatting, but also show errors that generally don't prevent the code from running.

**Comment by Starexify** [comment]
I personally use IntelliJ with the Haxe plugin and added haxeToolkit libs for autocompletion and code reference/docs
Tho it has limitations and the autocomplete doesn't work with multi classe files
The formatter is a bit wacky with vari but nothing too noticeable

I can also run the Funkin client in Terminal with a button

**Comment by TechnikTil** [comment]
https://marketplace.visualstudio.com/items?itemName=Snirozu.funkin-script-autocompleter

**Comment by Lasercar** [comment]
In your settings.json for your funkin project:
```json
  // Fix file associations for HScript.
  "files.associations": {
    "*.hxp": "haxe",
    "*.hscript": "haxe",
    "*.haxe": "haxe",
    "*.hxs": "haxe",
    "*.hxc": "haxe"
  },
  ```
The `files.associations` as seen above should have that last line. If you don't have it, add it! You'll also need the haxe language support extension ofc.

Also: https://github.com/FunkinCrew/Funkin/issues/4147
